client/state/sites/products/actions.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+
+import debugFactory from 'debug';
+import { mapValues } from 'lodash';
+import i18n from 'i18n-calypso';
+
+const debug = debugFactory( 'calypso:site-products:actions' );
+
+/**
+ * Internal dependencies
+ */
+import { createSiteProductObject } from './assembler';
```

Comment:
Interesting; this is my first time seeing this assembler pattern. Is this something we created in Calypso?

Reply:
I guess so -- I basically just copied this from https://github.com/Automattic/wp-calypso/blob/master/client/state/sites/plans/actions.js (i.e. the equivalent code for plans).